### DISCUSSION TITLE
Silhouette Portrait 3 Support

### DISCUSSION THREAD

**Discussion by Jeinzi** [Jeinzi/SMTCut#1] [General]
@mikhailmihalkov Let's continue the discussion here, rather than on a [random commit](https://github.com/Jeinzi/SMTCut/commit/f9289c431bf32565822c20da9ac6ebeefbeeb79a) nobody will see :)

I've uploaded [your changes](https://github.com/Jeinzi/SMTCut/commit/296a59bfd7da9cf66e46186917db6218b133892c) to a new branch, so we can discuss it in more detail :) I think I also need to have a look at the projects mentioned [here](https://github.com/pmonta/gerber2graphtec/issues/21#issuecomment-1047747204) and in the following comments. And I'll test if your modified program still works with my Silhouette Portrait 1.

**Comment by mikhailmihalkov** [comment]
Great) but I did it exclusively for portrait 3 with cameo 1 and the old version worked. Today I tested your code and for some reason the portrait 3 cuts with a turn of 180 degrees relative to the gerber fail... Do you have that? Trying to turn it in gerbbv does not give a result...

**Reply by Jeinzi** [reply]
So your saying that my current version rotates by 180°, but the old version does not?

**Reply by mikhailmihalkov** [reply]
Yes, but this is only on ubuntu 22.04 and with portrait 3 with cameo 1 there are no such problems... If you flip the file 180 degrees and save it to gerbbv, then when converted through a script, it turns back.. With win 10 works and converts correctly.... I assume it's about the converters

**Comment by mikhailmihalkov** [comment]
Checked on win10 does not work gives an error
running-win command line: gswin32cgswin32c.exe -q -dDELAYBIND -dWRITESYSTEMDICT -dNODISPLAY -dNOEPS C:/Users/MIKHAI~1/AppData/Local/Temp/psin2
Interpreter failure: -1
PostScript/PDF Interpreter finished. Return status -1 executed command : gswin32c -q -dDELAYBIND -dWRITESYSTEMDICT -dNODISPLAY -dNOEPS C:/Users/MIKHAI~1/AppData/Local/Temp/psin2
The interpreter seems to have failed, cannot proceed !

**Reply by Jeinzi** [reply]
I also tried it today on Windows, I encountered the same error :/

**Reply by mikhailmihalkov** [reply]
Figured out started cutting (cameo 1) on win 10 need to update pstoedit to version 3.78

**Reply by mikhailmihalkov** [reply]
win10\64 +pstoedit 3.78\64+Ghostscript 9.56.1\64+rawprint+cameo1+  Jeinzi/SMTCut ) = works

**Reply by mikhailmihalkov** [reply]
http://www.columbia.edu/~em36/windowsrawprint.html

**Reply by Jeinzi** [reply]
Thank you, also saw your other comment - after upgrading from pstoedit 3.71 to 3.78 it works for me, too.
What does not work is actually sending the file to the cutter. Do you use the GUI and if so, what did you type in the "Cutter Device Name" field? And did you install any driver for the cutter? Or did you use the windowsrawprint stuff?

**Reply by mikhailmihalkov** [reply]
Do not send directly from gui to printer in win 10. It worked in win 7 but not win 10. You need to create a network print in win 10 specify your usb printer port and select the Generic/Text Only driver to make it the default for convenience. Then you send the resulting txt file to rawprint and already send it directly to the printer. In the gui, you can register the path to rawprint and you can just send the ready-made file result.txt rawprint

**Reply by mikhailmihalkov** [reply]
1)Result.txt-2)rawrpint.exe-3) usb port (cameo1 driver Generic/Text Only)

**Reply by mikhailmihalkov** [reply]
I ask for petitions for my English)) I write through the translator)

**Reply by Jeinzi** [reply]
No worries about the language, I'm glad you're helping :) I'll try your Windows instructions at the end of the week, thank you :)

**Reply by mikhailmihalkov** [reply]
Unsubscribe by result)

**Comment by mikhailmihalkov** [comment]
Figured out started cutting (cameo 1) on win 10 need to update pstoedit to version 3.78

**Reply by Jeinzi** [reply]
Thank you, that helped a lot :)

**Reply by mikhailmihalkov** [reply]
Sometimes it happens that the first time you do not start cutting, resending the result.txt to rawprint helps.

**Reply by mikhailmihalkov** [reply]
The archive contains a complete set for cutting on win 10.. You may still need NET Framework and visual c++ if not installed.
https://drive.google.com/file/d/1mJRxD1NEjN9U6Kf53tb-ASF-i5PPWcpL/view?usp=sharing
https://gerbv.github.io/#download

**Reply by mikhailmihalkov** [reply]
[gerbv-win-static_20160713.zip](https://github.com/Jeinzi/SMTCut/files/9558048/gerbv-win-static_20160713.zip)

**Reply by Jeinzi** [reply]
Gerbv is working for me on Windows, ghostscript and pstoedit as well - the issue is just sending the file to the printer with the GUI. I actually managed to use file2graphtec.py (send2cutter.py) successfully, after following the instructions in the [robocut](https://github.com/Timmmm/robocut) readme and installing a driver via [Zadig](https://zadig.akeo.ie).

**Reply by mikhailmihalkov** [reply]
Excellent) By the way, checked the new version of gerbbv gives errors when checking the file rolled back to 2.6

**Reply by Jeinzi** [reply]
You mean `Unknown RS-274X extension found %TO% at line...` and variations thereof?
If so, I don't know what the problem is exactly, but it works despite the warnings. I was planning on opening an issue in the gerbv repo and looking into the export options in KiCad.

**Reply by mikhailmihalkov** [reply]
Yes, this error appears and if you install the old version of gerbbv 2.6, then there is no error. The second error appears if you need to rotate gerber to cut 90/180 degrees it does not always rotate

**Reply by mikhailmihalkov** [reply]
Jeinzi You can describe the uses in more detail
Zadig drivers? Win 10 win 7 ? What settings did you use after installing this driver? Creating a printer, etc.?

**Comment by Jeinzi** [comment]
I tried your version (https://github.com/Jeinzi/SMTCut/commit/296a59bfd7da9cf66e46186917db6218b133892c) with my Silhouette Portrait 1 last Friday; unfortunately it did not work and the instructions you commented are indeed needed. I'll have a look at how other projects deal with this.

**Reply by mikhailmihalkov** [reply]
My script will not work with portrait 1-2 and cameo 1-2. In the new models at the firmware level, the control commands changed a little... It should work with portrait 3 and cameo 4 (to be checked)

**Reply by mikhailmihalkov** [reply]
You need to use your script and update all programs from my message

**Reply by mikhailmihalkov** [reply]
Jeinzi Tell me where can there be an error in the code? The problem is that no matter how many degrees I turn gerber file in gerbv (180/90/270) Portrait 3 stubbornly cuts from 90 degrees angles relative to the fiducial. Where in the code is this parameter?

**Reply by mikhailmihalkov** [reply]
Jeinzi If you look at the resulting pdf, there it flips in the desired direction but the plotter cuts with a clockwise rotation angle of 90 degrees

**Reply by Jeinzi** [reply]
I don't know what exactly happens, I'm new to this codebase as well. The only thing that comes to mind is the orientation parameter in graphtec.py: In the [current master version](https://github.com/Jeinzi/SMTCut/blob/c8bd179aa8bad5ee21c51b2565576e24d7ffd146/graphtec.py#L38), it is set to 1. In [your version](https://github.com/Jeinzi/SMTCut/blob/296a59bfd7da9cf66e46186917db6218b133892c/graphtec.py#L35), it is set to 0.

[This documentation](https://github.com/fablabnbg/inkscape-silhouette/blob/main/Commands.md) on the graphtec commands says that FN0 means portrait mode and FN1 means landscape.

**Reply by mikhailmihalkov** [reply]
In this case, it turns 90 degrees or 180.... Where to find a fresh plotter command library is not clear

**Reply by mikhailmihalkov** [reply]
I changed the file a little and now it is enough to specify the path to rawprint.exe checked on win 10\64
[smtcut-gui.zip](https://github.com/Jeinzi/SMTCut/files/9617627/smtcut-gui.zip)
